README.md

```diff
@@ -20,463 +21,441 @@ This project now supports Swift 2, if you still require Swift 1.2 you can use th
 
-There is also the option for Dynamic Channels just by getting the channel with a name.
 
+## Channels
+ 
+ **`exp.getChannel(name, system, consumerApp)`** 
+ 
+ Returns a channel with the given name with two flags: `consumerApp` and `system`. Consumer devices can only listen and broadcast on consumer channels. System channels are listen only and can receive broadcasts about system events.
+ 
+```swift
+    let channel = ExpSwift.getChannel("my-channel",system: false,consumerApp: true)
+```
 
+**`channel.broadcast(name, payload, timeout)`** 
 
-### How to get channels
-```swift
-var orgchannel = ExpSwift.getChannel(SOCKET_CHANNELS.ORGANIZATION) as! OrganizationChannel 
-var systemChannel = ExpSwift.getChannel(SOCKET_CHANNELS.SYSTEM) as! SystemChannel
+Sends a broadcast with given `name` and `payload` on the channel. Waits for responses for `timeout` milliseconds and resolves with an array of responses.
 
-```
-###Dynamic Channels
-You can get your own channel by name with 
 ```swift
-var myChannel:CommonChannel = ExpSwift.getChannel("myChannel")
+    var payload:Dictionary<String,AnyObject> = ["opening":"knock knock?"]
+    channel.broadcast("hi", payload: payload1, timeout: "2000").then { result -> Void in
+        debugPrint(result)
+    }
 ```
 
-###  [Channel].fling(uuid)
-Fling content on a channel. UUID is the UUID of the content object you are flinging.
-```swift
-//FLING CONTENT
- var organizationChannel = ExpSwift.getChannel(SOCKET_CHANNELS.ORGANIZATION) as! OrganizationChannel
-     organizationChannel.fling("8930ff64-1063-4a03-b1bc-33e1ba463d7a")
-```
+**`channel.listen(name, callback)`** 
+
+Registers a [listener](#listeners) callback for events on the channel with the given `name`. Resolves to a [listener](#listeners) when the callback is registered and the network connection has subscribed to the channel.
+
+The callback is called with the broadcast payload as the first argument and a `respond` method as the second argument. Call the `respond` method to send a response back to the broadcaster.
 
-###  [Channel].listen(options, callback)
-Register a callback for a message on this channel.
-```swift
-//LISTEN FOR BROADCAST MESSAGE
-var orgchannel = ExpSwift.getChannel(OCKET_CHANNELS.ORGANIZATION) as! OrganizationChannel
-            var msg1:Dictionary<String,AnyObject> = ["name": "testing"]
-            orgchannel.listen(msg1,  callback: { (resultListen) -> Void in
-                            println(resultListen)
-            })
-```
-### [Channel].broadcast(options)
-Broadcast a message out on this channel. 
 ```swift
- //SEND BROADCAS MESSAGE
-var orgchannel = ExpSwift.getChannel(SOCKET_CHANNELS.ORGANIZATION) as! OrganizationChannel
-            var payload:Dictionary<String,String> = ["opening":"knock knock?"]
-            var msg2:Dictionary<String,AnyObject> = ["name": "testing","payload":payload]
-            orgchannel.broadcast(msg2)
+    channel.listen("myEvent",  callback: { (resultListen) -> Void in
```

Comment:
probably would be a good idea to show an example of the the respond method ie. respond("thank you")